GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -275,6 +294,70 @@ private void loadTagInfo(string revision)
             this.InvokeAsync(updateText);
         }
 
+
+        private void loadCommitSignInfo(GitRevision revision)
+        {
+            /* Good commit signature */
+            if (revision.SignInfo == "G")
+            {
+                commitSignPicture.Image = GitUI.Properties.Resources.certificate_ok;
+            }
+            else /* Any kind of not good commit signature */
+            {
+                commitSignPicture.Image = GitUI.Properties.Resources.certificate_error;
+            }
+        }
+
+
+        private int getNumberOfTagHere(GitRevision revision)
+        {
+            int nTagHere = 0;
+
+            foreach (GitRef gitRef in revision.Refs)
+            {
+                if (gitRef.IsTag && gitRef.IsDereference)
+                {
+                    nTagHere++;
+                }
+            }
+
+            return nTagHere;
+        }
+
+        private void loadTagSignInfo(GitRevision revision)
+        {
+            tagGpgMessage = "";
+            int tagCnt = 0;
+
+            foreach (GitRef gitRef in revision.Refs)
+            {
+                if (gitRef.IsTag && gitRef.IsDereference)
+                {
+                    tagGpgMessage = $"{tagGpgMessage}{gitRef.LocalName}\r\n{Module.GetTagGpgRawVerificationMessage(gitRef.LocalName)}\r\n\r\n";
+
+                    Regex tagRegex = new Regex("Good signature");
```

Comment:
I'll change it.
What do you mean with "precompile option" ? Is it something to be set?
Thanks.